src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java

```diff
@@ -1,54 +1,56 @@
 import seedu.trackbeau.model.tag.Tag;
+import seedu.trackbeau.commons.util.StringUtil;
 
 /**
  * Tests that a {@code Customer's}'s {@code Data} matches any of the keywords given.
  */
 public class SearchContainsKeywordsPredicate implements Predicate<Customer> {
-    private final String searchArea;
-    private final Integer searchType;
-    private final List<String> keywords;
+    private final ArrayList<List<String>> keywordsList;
 
     /**
      * Constructs a {@code Predicate}.
      *
-     * @param searchArea Data field of customer to search at.
-     * @param searchType Search for tags if true or keywords if false.
-     * @param keywords User input keywords to search with.
+     * @param keywordsList User input keywords to search with.
      */
-    public SearchContainsKeywordsPredicate(String searchArea, Integer searchType, List<String> keywords) {
-        this.searchArea = searchArea;
-        this.searchType = searchType;
-        this.keywords = keywords;
+    public SearchContainsKeywordsPredicate(ArrayList<List<String>> keywordsList) {
+        this.keywordsList = keywordsList;
     }
 
     @Override
     public boolean test(Customer customer) {
+        String[] find = {"getName", "getPhone", "getEmail", "getAddress", "getSkinType",
+                "getHairType", "getStaffs", "getServices", "getAllergies"};
         String searchString = "";
         Boolean anyMatch = false;
+
         try {
-            if (searchType == 0) {
-                searchString = customer.getClass().getDeclaredMethod(searchArea).invoke(customer).toString();
-            } else {
-                // Will always return type Set<Tag> from the 3 possible methods in the Customer class.
-                @SuppressWarnings("unchecked")
-                Set<Tag> tagList = (Set<Tag>) customer.getClass().getDeclaredMethod(searchArea).invoke(customer);
-                for (Tag tag : tagList) {
-                    searchString = searchString + tag.tagName + " ";
+            for (int i = 0; i < 9; i++) {
+                List<String> keywords = keywordsList.get(i);
+                if (i < 6) {
+                    searchString = customer.getClass().getDeclaredMethod(find[i]).invoke(customer).toString();
+                } else {
+                    // Will always return type Set<Tag> from the 3 possible methods in the Customer class.
+                    @SuppressWarnings("unchecked")
+                    Set<Tag> tagList = (Set<Tag>) customer.getClass().getDeclaredMethod(find[i]).invoke(customer);
+                    for (Tag tag : tagList) {
+                        searchString = searchString + tag.tagName + " ";
+                    }
                 }
-            }
-
-            for (String keyword : keywords) {
-                if (StringUtil.containsWordIgnoreCase(searchString, keyword)) {
-                    anyMatch = true;
-                    break;
+                if (keywords != null) {
```

Comment:
I feel lines 47-53 has arrow-head style code, maybe you can try doing 
`if (keywords == null) { 
      continue;
  }`
  `for (String keyword : keywords) {
      ...
  }
`

Reply:
okay will do